LLVM Baseline Survey 20250724

chenx97 opened the pull request:
Topic Description
-----------------

- pocl: rebuild for llvm baseline bump
    ... and use bash arrays in \`defines\` when possible
- opencl-clang: update to 20.1.0
- qtcreator: rebuild for llvm baseline bump
- pyside6: rebuild for llvm baseline bump
- qt-6: use llvm-20
- pyside2: rebuild for llvm baseline bump
- qt-5: use llvm-runtime-20
    - stop using ld.gold
    - use a different ENVREQ for arm64
- ldc: use llvm-18
    Updates shall be performed elsewhere
- mesa: update to 25.1.6
- spirv-llvm-translator: update to 20.1.4

... and 21 more commits

Package(s) Affected
-------------------

- adaptivecpp: 25.02.0-2
- apache-arrow: 19.0.1-1
- bcc: 0.35.0
- bpftrace: 0.23.5
- castxml: 0.6.11
- ccls: 0.20241108
- clazy: 1.15
- gnome-builder: 42.1-8
- ispc: 1.27.0-1
- kdevelop: 23.08.5-3
- ldc: 1.38.0-3
- libbpf: 1.6.1
- liblphobos: 1.38.0-3
- llvm: 1:20
- llvm-18: 18.1.8-11
- llvm-20: 20.1.8
- llvm-runtime: 1:20
- llvm-runtime-18: 18.1.8-11
- llvm-runtime-20: 20.1.8
- mesa: 1:25.1.6+dxheaders1.616.0
- opencl-clang: 20.1.0
- pocl: 1:7.0-1
- postgresql-13: 13.21-1
- postgresql-14: 14.18-1
- postgresql-15: 15.13-1
- postgresql-16: 16.9-1
- postgresql-17: 17.5-1
- postgresql-runtime-13: 13.21-1
- postgresql-runtime-14: 14.18-1
- postgresql-runtime-15: 15.13-1
- postgresql-runtime-16: 16.9-1
- postgresql-runtime-17: 17.5-1
- pyside2: 5.15.16-1
- pyside6: 6.8.2.1-3
- qt-5: 1:5.15.16+webengine5.15.18+webkit5.212.0+kde20241121-5
- qt-5-doc: 1:5.15.16+webengine5.15.18+webkit5.212.0+kde20241121-5
- qt-6: 6.8.2-10
- qt-6-doc: 6.8.2-10
- qtcreator: 16.0.1-3
- spirv-headers: 1:1.4.321.0
- spirv-llvm-translator: 20.1.4
- spirv-tools: 1:1.4.321.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit llvm-18 llvm-20 llvm adaptivecpp apache-arrow libbpf bcc bpftrace castxml ccls clazy gnome-builder ispc kdevelop postgresql-13 postgresql-14 postgresql-15 postgresql-16 postgresql-17 spirv-headers spirv-tools spirv-llvm-translator mesa liblphobos:+stage2 ldc qt-5 pyside2 qt-6 pyside6 qtcreator opencl-clang pocl
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [ ] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

chenx97 committed:
llvm: update to 20

chenx97 committed:
llvm-18: lower priority back to 50

The default LLVM will be 20.

chenx97 committed:
llvm-20: update to 20.1.8

... and raise priority for becoming the new default llvm

chenx97 committed:
kdevelop: use breezy instead of bzr

chenx97 committed:
bpftrace: update to 0.23.5

chenx97 committed:
bcc: update to 0.35.0

chenx97 committed:
adaptivecpp: rebuild for llvm baseline bump

... and remove BUILDDEP as every entry is also found in PKGDEP

chenx97 committed:
castxml: update to 0.6.11

chenx97 committed:
postgresql-14: rebuild for llvm-20

chenx97 committed:
gnome-builder: rebuild for llvm-20

chenx97 committed:
ccls: update to 0.20241108 and bump llvm to 19

The support for llvm-20 didn't arrive before the upstream tagged this version

chenx97 committed:
postgresql-17: rebuild for llvm-20

chenx97 committed:
postgresql-15: rebuild for llvm-20

chenx97 committed:
postgresql-16: rebuild for llvm-20

chenx97 committed:
postgresql-13: rebuild for llvm-20

chenx97 committed:
ispc: rebuild for llvm-20

chenx97 committed:
qt-6: use llvm-20

chenx97 committed:
pyside6: rebuild for llvm baseline bump

chenx97 committed:
apache-arrow: use llvm-18

... because reverse dependencies don't support the newer version

chenx97 committed:
qt-5: use llvm-runtime-20

- stop using ld.gold
- use a different ENVREQ for arm64

chenx97 committed:
clazy: update to 1.15

chenx97 commented:
This is a flight rn because ldc-related updates are skipped here. @liushuyu will handle them.

chenx97 committed:
libbpf: update to 1.6.1

chenx97 committed:
spirv-headers: update to 1.4.321.0

chenx97 committed:
spirv-tools: update to 1.4.321.0

chenx97 committed:
mesa: update to 25.1.6

chenx97 committed:
spirv-llvm-translator: update to 20.1.4

chenx97 committed:
pyside2: rebuild for llvm baseline bump

chenx97 committed:
ldc: use llvm-18

Updates shall be performed elsewhere

chenx97 commented:
ldc will be rebuilt instead of upgraded here. Removing the flight status.

SherkeyXD commented:
`git-interactive-rebase-tool` should use llvm-20

chenx97 commented:
> `git-interactive-rebase-tool` should use llvm-20

We are upgrading llvm to avoid having to do this.

chenx97 commented:
<img width="1535" height="900" alt="屏幕截图_20250801_103855" src="https://github.com/user-attachments/assets/63736c2e-a7d2-4caa-9f58-aa2bc1e439d7" />
Mesa works.

chenx97 commented:
<img width="429" height="715" alt="image" src="https://github.com/user-attachments/assets/3995afec-6a18-456a-ab07-d4bf1c6ffa76" />

```
sudo bpftrace -e "kprobe:tcp_*  /pid == `pidof Telegram`/ { printf(\"Entered: %s\n\", probe); }"
```
bpftrace LGTM

MingcongBai commented:
Clang-Tidy works with KDevelop:

<img width="3839" height="2159" alt="图片" src="https://github.com/user-attachments/assets/c9fbdc24-9d8b-436f-8cc5-61e538b068b8" />

MingcongBai commented:
Qt Creator fails:

```
qtc.extensionsystem.plugin: [ ClangFormat ] Plugin error: /usr/lib/qtcreator/plugins/libClangFormat.so: 无法加载库 /usr/lib/qtcreator/plugins/libClangFormat.so：libclang-cpp.so.18.1: 无法打开共享目标文件: 没有那个文件或目录
```

chenx97 committed:
qtcreator: rebuild for llvm baseline bump

- bump the REL twice because of a small mistake
- disable clang on riscv64 because of a mysterious ICE

chenx97 committed:
opencl-clang: update to 20.1.0

chenx97 committed:
pocl: rebuild for llvm baseline bump

- use bash arrays in `defines` when possible
- enable cl_khr_fp16 when possible (unless llvm-20 cannot do fp16 on your CPU)
